app/src/main/java/me/quadphase/qpdex/databaseAccess/PokemonFactory.java

```diff
@@ -892,7 +914,7 @@ public void setCaught(int nationalID, boolean value){
      */
     public int getCurrentGeneration() {
```

Comment:
this method does what getMaxGeneration() does, and should therefore be safely deleted.